### PR TITLE
feat: remove supports for legacy proof type

### PR DESCRIPTION
**Pull request by yutianwu**
### Description

This pr removes the supports for the legacy proof type. The ics23 proof is the only proof type supported.

### Rationale

There will be no legacy proof from the binance beacon chain any longer. Removing the supports for the legacy proof type will reduce the possibility of attacks.

### Example

N/A

### Changes

Notable changes: 
* remove supports for legacy proof type

### CODE REVIEW

File: core/vm/contracts.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ var PrecompiledContractsLuban = map[common.Address]PrecompiledContract{
 	common.BytesToAddress([]byte{9}): &blake2F{},
 
 	common.BytesToAddress([]byte{100}): &tmHeaderValidate{},
-	common.BytesToAddress([]byte{101}): &iavlMerkleProofValidatePlanck{},
+	common.BytesToAddress([]byte{101}): &iavlMerkleProofValidateLuban{},
```
Comment (brilliant-lx) [inline core/vm/contracts.go:158]: should not modify `PrecompiledContractsLuban`, 
need to create a new `PrecompiledContractsPlato`

File: core/vm/contracts_lightclient.go [hunk 2]
```diff
@@ -183,6 +183,21 @@ func (c *iavlMerkleProofValidatePlanck) Run(input []byte) (result []byte, err er
 	return c.basicIavlMerkleProofValidate.Run(input)
 }
 
+type iavlMerkleProofValidateLuban struct {
```
Comment (brilliant-lx) [inline core/vm/contracts_lightclient.go:186]: not luban, it is `plato`

File: core/vm/contracts.go [hunk 3]
```diff
@@ -210,11 +230,16 @@ func init() {
 	for k := range PrecompiledContractsLuban {
 		PrecompiledAddressesLuban = append(PrecompiledAddressesLuban, k)
 	}
+	for k := range PrecompiledContractsPlato {
+		PrecompiledAddressesLuban = append(PrecompiledAddressesLuban, k)
```
Comment (brilliant-lx) [inline core/vm/contracts.go:234]: typo?

File: core/vm/contracts_lightclient.go [hunk 4]
```diff
@@ -183,6 +183,21 @@ func (c *iavlMerkleProofValidatePlanck) Run(input []byte) (result []byte, err er
...
+type iavlMerkleProofValidatePlato struct {
+	basicIavlMerkleProofValidate
+}
+
+func (c *iavlMerkleProofValidatePlato) RequiredGas(_ []byte) uint64 {
+	return params.IAVLMerkleProofValidateGas
+}
+
+func (c *iavlMerkleProofValidatePlato) Run(input []byte) (result []byte, err error) {
+	c.basicIavlMerkleProofValidate.proofRuntime = v1.Ics23ProofRuntime()
```
Comment (guagualvcha) [inline core/vm/contracts_lightclient.go:195]: can we still enable:
```
	c.basicIavlMerkleProofValidate.verifiers = []merkle.ProofOpVerifier{
		forbiddenAbsenceOpVerifier,
		singleValueOpVerifier,
		multiStoreOpVerifier,
		forbiddenSimpleValueOpVerifier,
	}
```

**Review by guagualvcha — APPROVED** [review]

File: core/vm/contracts.go [hunk 5]
```diff
@@ -231,7 +231,7 @@ func init() {
 		PrecompiledAddressesLuban = append(PrecompiledAddressesLuban, k)
 	}
 	for k := range PrecompiledContractsPlato {
-		PrecompiledAddressesLuban = append(PrecompiledAddressesLuban, k)
+		PrecompiledAddressesLuban = append(PrecompiledAddressesPlato, k)
```
Comment (brilliant-lx) [inline core/vm/contracts.go:234]: typo: PrecompiledAddressesLuban

**Review by brilliant-lx — APPROVED** [review]

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**